### PR TITLE
Components: Allow extra props for RadioControl component

### PR DESCRIPTION
**Pull request by ser-manjeet**
## Description

This PR allows extra props for RadioControl component like `disabled` or ARIA related props.

## Types of changes

It's an enhancement that aligns the component's API with other components.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by gziolo** [comment]
What props are exactly missing? Should they be part of the public API maybe?

@ItsJonQ, how optional props are handled in G2 components? Ideally, it should be either all components take custom props or only explicitly defined props are allowed.

**Review by sarayourfriend — APPROVED** [review]
This LGTM, thanks for the contribution!

@gziolo I think there are basically two classes of components in WordPress components. @ItsJonQ and I have discussed this before a couple of times. There are "primitive" components, like `Button` and the various `Control`s and there are UI components (I have no idea what to call these, I've seen them called UI Blocks before but "blocks" is too overloaded in the Gutenberg context). Each of these have different considerations as far as prop forwarding goes. For the primitive components, I think we should forward props as this PR does because it allows the consumer to fully control the underlying component without requiring us to document each and every single DOM attribute. G2 does this via the context system (and we've even successfully type annotated this through the `PolymorphicComponent` type). On the other hand, UI components like `Placeholder`, `AnglePickerControl`, or `FontSizeControl` don't make sense to do this for because... well, where would we forward those props onto? I think in the case of those components, they're so specific that they should just be used as is, rather than the more general "primitive" components.

Hopefully that makes sense. This is definitely just something that @ItsJonQ and I came up with to explain our thought process as this was discussed during the FontSizeControl intregration (we narrowed the props in that case) so it's something work having an ongoing discussion about, especially around the language we use to denote each type of component we have in this library.

> What props are exactly missing? Should they be part of the public API maybe?

In thise case all DOM attribute props are missing (disabled, aria props, any additional data attributes the consumer may want to apply, etc). To document this as part of the API we merely have to add `& React.HTMLProps<'div'>` to the final prop type for this component.

File: packages/components/src/radio-control/index.js [hunk 1]
```diff
@@ -53,6 +54,7 @@ export default function RadioControl( {
 							aria-describedby={
 								!! help ? `${ id }__help` : undefined
 							}
+							{...props}
```
Comment (gziolo) [inline packages/components/src/radio-control/index.js:57]: Linter complain about this line:

> [1] /home/runner/work/gutenberg/gutenberg/packages/components/src/radio-control/index.js
78
> [1]   57:9  error  Replace `...props` with `·...props·`  prettier/prettier


```suggestion
							{ ...props }
```

**Comment by gziolo** [comment]
@saramarcondes – thank you for your detailed response. I agree with the reasoning. It all makes sense in the context of 3rd party usage. For those low-level/primitive components, it all makes sense to allow any props to be passed as an advanced use case. 

> > What props are exactly missing? Should they be part of the public API maybe?
> In thise case all DOM attribute props are missing (disabled, aria props, any additional data attributes the consumer may want to apply, etc). To document this as part of the API we merely have to add & `React.HTMLProps<'div'>` to the final prop type for this component.

All of the reasons to add props you mentioned are very much specific to the web. Do we plan to make G2 components cross-platform? At the moment it looks like they are for usage with the web only. If the answer is yes, we should consider a translation layer for stuff that becomes problematic like styles or accessibility-related attributes. It probably should be its own discussion though 😄

**Comment by saramarcondes** [comment]
> All of the reasons to add props you mentioned are very much specific to the web. Do we plan to make G2 components cross-platform? At the moment it looks like they are for usage with the web only. If the answer is yes, we should consider a translation layer for stuff that becomes problematic like styles or accessibility-related attributes. It probably should be its own discussion though 😄

Shall we start an issue to discuss this? I don't know what the plan is for this at all.

cc @ItsJonQ

**Comment by gziolo** [comment]
It all depends on the mobile team if they plan to invest time in making G2 components cross-platform. Regardless, it probably is s good idea to open an issue to have this discussion well documented.

**Comment by ItsJonQ** [comment]
> If the answer is yes, we should consider a translation layer for stuff that becomes problematic like styles or accessibility-related attributes

@gziolo As @saramarcondes had mentioned, I'm not sure what the plan is for this.

However, I do feel strongly that Component library UI should provide the ability to render attributes accepted by HTML (e.g. `data-*`) and to extends props regularly used in React (e.g. `className`, `style`). Not supporting this often results in attempts to insert attributes or render styles in alternative less-desirable ways.

Examples may include:

Adding fragile styles in the tune of `.container-component > div`. Or worse... Creating a `ref`, and adding attributes directly to the DOM node.

As mentioned, this does mean that web props may not translate well into React Native.
We have something working in our favour though (from the G2 Components side)...

`create-core-element` - The thing that creates a `View`.

During this process, props are filtered out:

https://github.com/ItsJonQ/g2/blob/eb8deab0d64cfe2930b59494a8d3c39fb07ebef2/packages/create-styles/src/create-style-system/create-core-element.js#L130

This implementation is based on Emotion, and is a standard process for CSS-in-JS solutions. Interesting, it appears that other libraries have standardized around using `@emotion/is-prop-valid`.

For RN, we could adjust this to accommodate RN components - either remapping certain properties or filtering them out entirely.

> Regardless, it probably is s good idea to open an issue to have this discussion well documented.

Sounds good :)

**Comment by jeffpaul** [comment]
@ser-manjeet I've tried to reach out via other channels, but I have been unsuccessful in getting in touch. I wanted to make sure you saw the request to you for consent to re-license your contributions to Gutenberg under GPLv2 and MPLv2 here: https://github.com/WordPress/gutenberg/issues/31893. If you would kindly review that description and comment accordingly it would be greatly appreciated.

Thanks!
Jeff.

**Pull request merged**